cmd/object-handlers.go

```diff
@@ -1365,7 +1365,7 @@ func (api objectAPIHandlers) NewMultipartUploadHandler(w http.ResponseWriter, r
 	var encMetadata = map[string]string{}
 
 	if objectAPI.IsEncryptionSupported() {
-		if hasServerSideEncryptionHeader(r.Header) {
+		if crypto.IsRequested(r.Header) {
```

Comment:
validation needs to add a condition to reject sse-copy source headers if present, since crypto.IsRequested() is more generic now.
```
// X-Amz-Copy-Source shouldn't be set for this call.
	if _, ok := r.Header[xhttp.AmzCopySource]; ok {
		writeErrorResponse(ctx, w, errorCodes.ToAPIErr(ErrInvalidCopySource), r.URL, guessIsBrowserReq(r))
		return
	}
```

Reply:
Same as https://github.com/minio/minio/pull/8276/files/1ccbd16f754072c440d32f21d62be7c973b95626#r326426213